### ISSUE TITLE
Missing includes with recent versions of gcc

### ISSUE THREAD

**Issue by mbanck** [nubakery/bagel#73]
Similar to what was fixed in 8baa305852cfb1d5fbf2d61669da9d1c45f4abe0 I see the following build failure with Debian's unstable gcc (gcc version 5.3.1 20160429 (Debian 5.3.1-17) as of current master (c1e750cbc9414d074827b7d94b9ebab784f885cf):

```
/bin/bash ../../libtool  --tag=CXX   --mode=compile mpicxx -DHAVE_CONFIG_H -I. -I../..   -std=c++11  -I../.. -g -O2 -DNDEBUG -std=c++11 -c -o comprys/_complex_eriroot_1.lo comprys/_complex_eriroot_1.cc
libtool: compile:  mpicxx -DHAVE_CONFIG_H -I. -I../.. -std=c++11 -I../.. -g -O2 -DNDEBUG -std=c++11 -c comprys/_complex_eriroot_1.cc -o comprys/_complex_eriroot_1.o
comprys/_complex_eriroot_1.cc: In static member function 'static void bagel::ComplexERIRootList::complex_eriroot1(const std::complex<double>*, std::complex<double>*, std::complex<double>*, int)':
comprys/_complex_eriroot_1.cc:1268:11: error: 'vector' was not declared in this scope
           vector<double> xrval(10);
           ^
comprys/_complex_eriroot_1.cc:1268:18: error: expected primary-expression before 'double'
           vector<double> xrval(10);
                  ^
```

Including ```<vector>``` in ```src/integral/comprys/comperirootlist.h```, ```src/integral/ecp/sphusplist.h``` and ```<numeric>``` in ```src/smith/storage.h``` fixes those kind of issues for me.

**Comment by shiozaki** [comment]
Ryan, this is your code; please fix and close. In addition, vector should be replaced by array\<double,10\> for a performance reason.

**Comment by shiozaki** [comment]
PS: I do not see anything that is dependent on the "numeric" header in storage.h, so the suggested fix is not the right approach.

**Comment by mbanck** [comment]
The failure I saw there was:

```
bin/bash ../../libtool  --tag=CXX   --mode=compile mpicxx -DHAVE_CONFIG_H -I. -I../..   -std=c++11  -I../.. -g -O2 -DNDEBUG -std=c++11 -c -o queue.lo queue.cc
libtool: compile:  mpicxx -DHAVE_CONFIG_H -I. -I../.. -std=c++11 -I../.. -g -O2 -DNDEBUG -std=c++11 -c queue.cc -o queue.o
In file included from ../../src/smith/task.h:33:0,
                 from ../../src/smith/queue.h:35,
                 from queue.cc:28:
../../src/smith/storage.h: In member function 'size_t bagel::SMITH::StorageIncore<DataType>::blocksize(const std::vector<bagel::SMITH::Index>&) const':
../../src/smith/storage.h:161:66: error: 'accumulate' is not a member of 'std'
     size_t blocksize(const std::vector<Index>& p) const { return std::accumulate(p.begin(), p.end(), 1lu, [](size_t a, const Index& b){ return a*b.size(); }); }
                                                                  ^
make[4]: *** [queue.lo] Error 1
Makefile:902: recipe for target 'queue.lo' failed
make[4]: Leaving directory '/«PKGBUILDDIR»/src/smith'
```

**Comment by shiozaki** [comment]
Ah Ok, thanks. I thought it's in algorithm.

**Comment by shiozaki** [comment]
Ryan @RDReynolds , please fix this ASAP

**Comment by shiozaki** [comment]
Should be fixed by a4f7469fbf4373cfb18b6239bdba12d51dfdf226
